File: java/interpreter/ej2/TerminalExpression_Minus.java
package patrones.interpreter.ej2;

import java.util.Stack;
/**
 * Expresion resta
 *
 */
public class TerminalExpression_Minus implements Expression {
	
	public void interpret(Stack<Integer> s) {
		int tmp = s.pop();
		s.push(s.pop() - tmp);
	}
}

File: c#/interpreter/Ejm2/TerminalExpression_Minus.cs
using System.Collections;
namespace Interpreter.Ejm2
{
    public class TerminalExpression_Minus : Expression
    {

        public void Interpret(Stack s)
        {
            int tmp = (int)s.Pop();
            s.Push((int)s.Pop() - tmp);
        }
    }
}
